eng/common/scripts/link-replacement.ps1

```diff
@@ -0,0 +1,54 @@
+param (
+  # url list to verify links. Can either be a http address or a local file request. Local file paths support md and html files.
+  [string] $sourceDir,
+  # file that contains a set of links to ignore when verifying
+  [string] $ignoreLinksFile = "$PSScriptRoot/ignore-links.txt",
+  # switch that will enable devops specific logging for warnings
+  [switch] $devOpsLogging = $false,
+  [string] $branchReplaceRegex = "(https://github.com/.*/(?:blob|tree)/)master(/.*)",
+  # the substitute branch name or SHA commit
+  [string] $releaseTag = ""
+)
+
+$regexOptions = [System.Text.RegularExpressions.RegexOptions]"Singleline, IgnoreCase";
+
+function ReplaceLink ($scanFolder, $fileSuffix, $replacement, $customRegex) {
+  $regex = new-object System.Text.RegularExpressions.Regex ($branchReplaceRegex, $regexOptions)
```

Comment:
Why are you creating a custom regex object?

Reply:
Copy-docs-blobstorage call the `ReplaceLink` function. Diff repo will pass different regex into the function.

Reply:
It is fine to pass in an regex string but you don't need to explicitly call new-object Regex, you can just use the string. Is there a reason you need to construct a RegEx object?